src/adapters/plugin/plugin.js

```diff
@@ -178,8 +188,50 @@ class Plugin {
     return this.ipcSocket.sendJson(msg);
   }
 
+  /**
+   * Does cleanup required to allow the test suite to complete cleanly.
+   */
+  shutdown() {
+    this.ipcSocket.close();
+  }
+
+  start() {
+    // If we need embedded spaces, then consider changing to use the npm
+    // module called splitargs
+    this.restart = true;
+    let args = this.exec.split(' ');
+    this.process = spawn(args[0], args.slice(1));
```

Comment:
I don't love that we're allowing add-ons to execute arbitrary commands on start-up. Is there any reason to not force them to be loaded by `adapter-loader.js`?

Reply:
adapter-loader.js is really only for node plugins. As soon as a plugin written in another language comes along you'll have the same problem.

Besides, what's to stop a plugin loaded via adapter-loader.js from calling spawn? If you really wanted to lock this down you'd need to do something like use seccomp.